packages/medusa-plugin-elasticsearch/src/services/elasticsearch.js

```diff
@@ -0,0 +1,84 @@
+import { SearchService } from "medusa-interfaces"
+import { Client } from "elasticsearch"
+import { indexTypes } from "medusa-core-utils"
+import { transformProduct } from "../utils/transform-product"
+
+class ElasticSearchService extends SearchService {
+    constructor(container, options) {
+        super()
+
+        this.options_ = options
+
+        this.client_ = new Client(options.config)
```

Comment:
ElasticSearch Client mutates the config when passed in raw (see [here](https://github.com/elastic/elasticsearch-js/issues/33)). This means, that ElasticSearch will complain the second time we try to init a client, since its two different objects.

Instead we should add a function:
```
getConfig() {
    return { ...this.options_.config }
}
```
And use this when initializing the function: `this.client_ = new Client(options.config)`

Hope this makes sense :)